### PR TITLE
Add a map of values_approx functions from cpu to gpu.

### PR DESCRIPTION
**Pull request by abergeron**
This should fix the problems in DEBUG_MODE.

### CODE REVIEW

**Comment by lamblin** [comment]
Jenkins seems to be full of the following message:
```
[Theano_GPU] ERROR (theano.gof.opt): Optimization failure due to: local_abstractconv_check
[Theano_GPU] ERROR (theano.gof.opt): node: AbstractConv2d{convdim=2, border_mode='valid', subsample=(1, 1), filter_flip=True, imshp=(10, 3, 5, 5), kshp=(12, 3, 3, 3), filter_dilation=(1, 1), num_groups=1, unshared=False}(GpuReshape{4}.0, GpuReshape{4}.0)
[Theano_GPU] ERROR (theano.gof.opt): TRACEBACK:
[Theano_GPU] ERROR (theano.gof.opt): Traceback (most recent call last):
[Theano_GPU]   File "/home/jenkins/workspace/Theano_PR/theano/gof/opt.py", line 2034, in process_node
[Theano_GPU]     replacements = lopt.transform(node)
[Theano_GPU]   File "/home/jenkins/workspace/Theano_PR/theano/tensor/nnet/opt.py", line 500, in local_abstractconv_check
[Theano_GPU]     node.op.__class__.__name__)
[Theano_GPU] LocalMetaOptimizerSkipAssertionError: AbstractConv2d Theano optimization failed: there is no implementation available supporting the requested options. Did you exclude both "conv_dnn" and "conv_gemm" from the optimizer? If on GPU, is cuDNN available and does the GPU support it? If on CPU, do you have a BLAS library installed Theano can link against? On the CPU we do not support float16.
```
I'm not sure where that's coming from.

**Comment by nouiz** [comment]
I'm ok with this PR. @abergeron, can you check if what @lamblin found was already there before this PR? If so, we could merge this PR now.

**Comment by abergeron** [comment]
That logspam from the tests is there since at least 0.9, probably even before.  It's not actually an error in the tests themselves.

File: theano/gpuarray/type.py [hunk 1]
```diff
@@ -542,6 +517,65 @@ def c_code_cache_version(self):
...
+                     rtol=None, atol=None):
+    if a.shape != b.shape or a.dtype != b.dtype:
+        return False
+    if str(a.dtype) in theano.tensor.discrete_dtypes:
+        return GpuArrayType.values_eq(a, b)
+    else:
+        if allow_remove_inf or allow_remove_nan:
+            raise NotImplementedError(
+                "GpuArrayType.values_eq_approx() don't implemented the"
+                " allow_remove_inf and allow_remove_nan parameter")
```
Comment (lamblin) [inline theano/gpuarray/type.py]: Maybe we only have to remove that error, and jump directly to the section after `# maybe the trouble is that there are NaNs`?

Comment (abergeron) [reply theano/gpuarray/type.py]: Yes, I think that could work.

**Pull request merged**